src/ru/itmo/kotlin/plugin/diagnosticRelated.kt

```diff
@@ -0,0 +1,116 @@
+package ru.itmo.kotlin.plugin
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.diagnostics.DiagnosticFactory0
+import org.jetbrains.kotlin.diagnostics.DiagnosticFactory1
+import org.jetbrains.kotlin.diagnostics.Errors
+import org.jetbrains.kotlin.diagnostics.Severity.ERROR
+import org.jetbrains.kotlin.diagnostics.Severity.WARNING
+import org.jetbrains.kotlin.diagnostics.SourceElementPositioningStrategies.NAME_IDENTIFIER
+import org.jetbrains.kotlin.diagnostics.SourceElementPositioningStrategies.OVERRIDE_MODIFIER
+import org.jetbrains.kotlin.diagnostics.error0
+import org.jetbrains.kotlin.diagnostics.rendering.DefaultErrorMessages
+import org.jetbrains.kotlin.diagnostics.rendering.DiagnosticFactoryToRendererMap
+import org.jetbrains.kotlin.diagnostics.rendering.Renderers.RENDER_CLASS_OR_OBJECT
+import org.jetbrains.kotlin.diagnostics.warning0
+import org.jetbrains.kotlin.diagnostics.warning1
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import ru.itmo.kotlin.plugin.StateLoggingErrors.ANCESTOR_WITH_ANNOTATION
+import ru.itmo.kotlin.plugin.StateLoggingErrors.DUPLICATING_ANNOTATIONS
+import ru.itmo.kotlin.plugin.StateLoggingErrors.GETTER_OVERRIDE_IS_NOT_ALLOWED
+import ru.itmo.kotlin.plugin.StateLoggingErrors.NO_LOGGER_CLASS_ANNOTATION
+import ru.itmo.kotlin.plugin.StateLoggingErrors.RETURN_HAS_NO_EFFECT
+import ru.itmo.kotlin.plugin.StateLoggingErrors.TARGET_DELEGATE_IS_NOT_ALLOWED
+import ru.itmo.kotlin.plugin.StateLoggingErrors.TARGET_IS_NOT_ALLOWED
+import ru.itmo.kotlin.plugin.StateLoggingErrors.TARGET_SHOULD_BE_CLASS
+import ru.itmo.kotlin.plugin.StateLoggingErrors.TARGET_SHOULD_NOT_BE_ENUM_CLASS
+
+
+object KtStateLoggingErrors {
+    val TARGET_SHOULD_BE_CLASS by error0<PsiElement>(NAME_IDENTIFIER)
+    val TARGET_SHOULD_NOT_BE_ENUM_CLASS by error0<PsiElement>(NAME_IDENTIFIER)
+    val GETTER_OVERRIDE_IS_NOT_ALLOWED by error0<PsiElement>(OVERRIDE_MODIFIER)
+    val NO_LOGGER_CLASS_ANNOTATION  by error0<PsiElement>(NAME_IDENTIFIER)
+    val TARGET_IS_NOT_ALLOWED  by error0<PsiElement>(NAME_IDENTIFIER)
+    val RETURN_HAS_NO_EFFECT by warning0<PsiElement>(NAME_IDENTIFIER)
+    val ANCESTOR_WITH_ANNOTATION  by warning1<PsiElement, PsiElement>(NAME_IDENTIFIER)
+}
+
+
+object StateLoggingErrors {
```

Comment:
`DiagnosticFactoryN` are used only for FE 1.0, so there is no need to declare them